design/name-system.md

```diff
@@ -0,0 +1,103 @@
+# Noosphere Name System
+
+The Noosphere Name System (NNS) is a distributed network, similar to [DNS], that resolves a sphere's identity to the latest address of a revision of the sphere.
+
+## Terms
+
+* **Spheres** are a core structure in Noosphere representing an entire "workspace" or "notebook", containing many documents, with change history represented as a series of revisions, not dissimilar from Git. A sphere's capabilities are managed by a public/private [Ed25519] encryption key. The sphere's unique identity is represented and publically referenced as a [DID]. 
+* **Decentralized Identifiers** ([DID]s) are public keys of a public/private key pair represented as a text string. The DIDs used in NNS are all derived from [Ed25519] encryption keys.
+  * Example representation: `did:key:z6MkoE19WHXJzpLqkxbGP7uXdJX38sWZNUWwyjcuCmjhPpUP`
+* **User Controlled Authorization Networks** ([UCAN]s) are an extension of JSON Web Tokens ([JWT], [RFC7519]), designed to enable ways of authorizing offline-first apps and distributed systems. Represented as JSON data, base64 encoded as a string.
+* **[Multiaddr]s** is an extensible format describing a network address through multiple layers and protocols.
+  * Example representation for a machine at IP address 10.20.30.1 with TCP port 20000: `/ip4/10.20.30.1/tcp/20000`
+* **[PeerId]s** are [multihash]ed public keys. In NNS, PeerIds are derived from keys unrelated to spheres, and used to identify and communicate with DHT nodes in the network.
+* **Content Identifiers** ([CID]s) are [multihash]ed [content addresses](https://docs.ipfs.tech/concepts/content-addressing/), where content can be identified by an immutable address. CIDs used in NNS are all CIDv1.
```

Comment:
Should we specify any specific hash type for CIDv1?

Reply:
We're using Blake2b256 for everything right now. We may not be able to get away with requiring a specific hash over the long run, but for now it saves us time to assume this hash everywhere.

Reply:
Perfect, will treat similarly as Ed25519-based DID keys (we use them, others may be supported but TBD).